### ISSUE TITLE
Compiling VM with Clang1000 on macOS 10.14 makes UI windows start zoomed in

### ISSUE THREAD

**Issue by erg** [factor/factor#2070]
The same ``factor.image`` and factor binary compiled with Clang900 show the correct behavior.

Happens still on 10.14.1 Beta (18B45d) with below clang.
```
% clang --version
Apple LLVM version 10.0.0 (clang-1000.11.45.2)
Target: x86_64-apple-darwin18.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by mrjbq7** [comment]
And (I think) the same ``factor.image`` with only the binary VM compiled with Clang1000 shows the buggy behavior.  Weird that the C recompile breaks it that way!

**Comment by jckarter** [comment]
It's possible that there's an SDK version check in the Apple frameworks somewhere to change behavior based on the Xcode version you compiled with. This is often how bug fixes or other behavior changes are staged in while providing compatibility with existing binaries.

**Comment by mrjbq7** [comment]
Maybe related?

https://github.com/wxWidgets/wxWidgets/pull/846

**Comment by catb0t** [comment]
seems wxWidgets was reporting crashes but nothing about scaling / zooming problems, but maybe the PR also fixed those unreported issues  

https://trac.wxwidgets.org/search?q=mojave

**Comment by mrjbq7** [comment]
Still buggy on 10.14.1.

**Comment by catb0t** [comment]
Is the bug present if you compile the VM and image with Clang1000?

**Comment by mrjbq7** [comment]
Only matters if you compile VM binary with clang1000. Not present with clang900.

**Comment by erg** [comment]
https://developer.apple.com/library/archive/documentation/GraphicsImaging/Conceptual/OpenGL-MacProgGuide/opengl_threading/opengl_threading.html

http://mirror.informatimago.com/next/developer.apple.com/qa/qa2001/qa1209.html

https://metashapes.com/blog/advanced-nsview-setup-opengl-metal-macos/


https://github.com/golang/exp/blob/master/shiny/driver/gldriver/cocoa.m#L281


https://stackoverflow.com/questions/20083027/nsopenglview-and-cvdisplaylink-no-default-frame-buffer

**Comment by erg** [comment]
Opening the window flashes the correct height and then goes to 2x height (wrong, too tall).

**Comment by erg** [comment]
Do we need to explicitly call ``backingScaleFactor``? This page warns against it.

https://developer.apple.com/documentation/appkit/nsscreen/1388385-backingscalefactor

**Comment by mrjbq7** [comment]
That might make it cleaner and allow support of non retina screens

**Comment by erg** [comment]
Fixed by f879a012c5184a09139adfbf78a7e9f7ffb116e7
